test/varity/vcf_lift_test.clj

```diff
@@ -0,0 +1,124 @@
+           {:failure [{:alt ["A"], :ref "AT", :pos 5, :filter [:PASS], :id nil,
+                       :info {:END 6}, :qual 100.0, :chr "chr1"}]}))))
+
+(deftest vcf-forward-lift-test
+  (with-open [vcf-forward-rdr
+              (vcf/reader "./test-resources/vcf-lift/sample-forward.vcf")
+              seq-reader
+              (io-seq/reader "./test-resources/vcf-lift/sample.fasta")]
+    (is (= (vcf-lift/liftover-variants
+            seq-reader
+            (ch/index (ch/load-chain
+                       "./test-resources/vcf-lift/sample-forward.chain"))
+            (vcf/read-variants vcf-forward-rdr))
+           {:success [{:alt ["AT"], :ref "A", :pos 5, :filter [:PASS],
+                       :id "1", :info {:END 5}, :qual 100.0, :chr "chr1"}
+                      {:alt ["ATA" "G"], :ref "AT", :pos 5, :filter [:PASS],
+                       :id "2", :info nil, :qual 100.0, :chr "chr1"}
+                      {:alt ["CG"], :ref "C", :pos 8, :filter [:PASS],
+                       :id "3", :info {:END 8}, :qual 100.0, :chr "chr1"}]
+            :failure nil}))))
+
+(deftest vcf-multi-lift-test
+  (with-open [vcf-forward-rdr
+              (vcf/reader "./test-resources/vcf-lift/sample-multi.vcf")
+              seq-reader
+              (io-seq/reader "./test-resources/vcf-lift/sample.fasta")]
+    (is (= (vcf-lift/liftover-variants
+            seq-reader
+            (ch/index (ch/load-chain
+                       "./test-resources/vcf-lift/sample-multi.chain"))
+            (vcf/read-variants vcf-forward-rdr))
+           {:failure nil,
+            :success [{:alt ["A"], :ref "ATGCATGC", :pos 9, :filter [:PASS],
+                       :id "1", :info {:END 16}, :qual 100.0, :chr "chr1"}
+                      {:alt ["A"], :ref "ATGCATGC", :pos 9, :filter [:PASS],
+                       :id "3", :info {:END 16}, :qual 100.0, :chr "chr1"}
+                      {:alt ["A"], :ref "ATGCATGCATGCATGCATGC", :pos 9,
+                       :filter [:PASS], :id "2", :info {:END 28}, :qual 100.0,
+                       :chr "chr1"}
+                      {:alt ["T"], :ref "TG", :pos 54, :filter [:PASS], :id "4",
+                       :info {:END 55}, :qual 100.0, :chr "chr1"}]}))))
+
+(deftest multi-hit-chain
+  (let [chain-index (ch/index (ch/load-chain "./test-resources/vcf-lift/sample-multi.chain"))]
+    (testing "descending score chain"
+      (is (= (#'vcf-lift/calc-new-interval "chr1" 178 179 chain-index)
+             {:chr "chr1", :start 58, :end 59, :strand :forward}))
+      (is (= (#'vcf-lift/calc-new-interval "chr1" 182 183 chain-index)
+             {:chr "chr1", :start 52, :end 53, :strand :forward}))
+      (is (= (#'vcf-lift/calc-new-interval "chr1" 191 192 chain-index)
+             {:chr "chr1", :start 51, :end 52, :strand :forward})))
+    (testing "ascending score chain"
+      (is (= (#'vcf-lift/calc-new-interval "chr4" 131 191 (ch/index (ch/load-chain "./test-resources/vcf-lift/sample-multi.chain")))
```

Comment:
`chain-index` can be reused for this test case as well, right?

Reply:
Thank you for pointing. I fixed it.